pandas/tests/io/test_stata.py

```diff
@@ -1374,7 +1374,7 @@ def test_unsupported_datetype(self):
                 "dates": dates,
             }
         )
-        with pytest.raises(NotImplementedError):
+        with pytest.raises(NotImplementedError, match=msg):
```

Comment:
```suggestion
        with pytest.raises(
            NotImplementedError,
            match=re.escape("Data type datetime64[ns, Asia/Hong_Kong] not supported."),
        ):
```

also re needs to be imported.